pulpcore/pulpcore/app/models/task.py

```diff
@@ -117,11 +133,27 @@ class Worker(Model):
 
     name = models.TextField(db_index=True, unique=True)
     last_heartbeat = models.DateTimeField(auto_now=True)
-    online = models.BooleanField(default=True)
+    gracefully_stopped = models.BooleanField(default=True)
 
-    def save_heartbeat(self):
-        """Save a worker heartbeat
+    @property
+    def online(self):
+        """
+        Whether a worker can be considered 'online'
+
+        To be considered 'online', a worker must have a recent heartbeat timestamp and must not
+        have the 'gracefully_stopped' flag set to True. "Recent" is defined here as "within
+        the pulp process timeout interval".
+
+        Returns:
+            bool: Whether the worker is online or not.
+        """
+        now = timezone.now()
+        age_threshold = now - timedelta(seconds=TASKING_CONSTANTS.PULP_PROCESS_TIMEOUT_INTERVAL)
+
+        return not self.gracefully_stopped and self.last_heartbeat > age_threshold
```

Comment:
Probably not an important aspect, but to be consistent w/ the inequality used in the manager method I think this should this be `>=`